storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/BlindedBlockMigration.java

```diff
@@ -107,4 +129,167 @@ private void moveFirstFinalizedBlockToBlindedStorage() {
+    final long preBellatrixMigratedBlocks =
+        migrateFinalizedBlocksPreBellatrix(finalizedBlockIndexCounter);
+    migrateFinalizedBlocks(finalizedBlockIndexCounter - preBellatrixMigratedBlocks);
+    migrateNonCanonicalBlocks();
+  }
+
+  private long copyFinalizedBlockIndexToBlindedStorage() {
+    long counter = 0;
+    try (final Stream<Map.Entry<Bytes32, UInt64>> blockRoots =
+        dao.streamUnblindedFinalizedBlockRoots()) {
+      for (Iterator<Map.Entry<Bytes32, UInt64>> it = blockRoots.iterator(); it.hasNext(); ) {
+        if (counter == 0) {
+          LOG.info("Copying finalized block indices to blinded storage");
+        }
+        try (KvStoreCombinedDaoBlinded.FinalizedUpdaterBlinded finalizedUpdaterBlinded =
+            dao.finalizedUpdaterBlinded()) {
+          for (int i = 0; i < INDEX_BATCH_SIZE && it.hasNext(); i++) {
+            final Map.Entry<Bytes32, UInt64> entry = it.next();
+            final Bytes32 root = entry.getKey();
+            final UInt64 slot = entry.getValue();
+            finalizedUpdaterBlinded.addFinalizedBlockRootBySlot(slot, root);
+            counter++;
+          }
+          finalizedUpdaterBlinded.commit();
+        }
+
+        pause();
+      }
+    }
+    if (counter > 0) {
+      LOG.info("{} finalized block indices copied", counter);
+    }
+    return counter;
+  }
+
+  private long migrateFinalizedBlocksPreBellatrix(final long migrationCounter) {
+    long counter = 0;
+    boolean preBellatrix = true;
+    if (migrationCounter > 0) {
+      LOG.info(
+          "Migrating pre-bellatrix blocks to blinded storage, {} finalized blocks to migrate",
+          migrationCounter);
+    }
+    try (final Stream<Map.Entry<Bytes, Bytes>> stream = dao.streamUnblindedFinalizedBlocksRaw()) {
+      for (Iterator<Map.Entry<Bytes, Bytes>> it = stream.iterator();
+          it.hasNext() && preBellatrix; ) {
+        try (KvStoreCombinedDaoBlinded.FinalizedUpdaterBlinded finalizedUpdaterBlinded =
+                dao.finalizedUpdaterBlinded();
+            KvStoreCombinedDaoUnblinded.FinalizedUpdaterUnblinded finalizedUpdaterUnblinded =
+                dao.finalizedUpdaterUnblinded()) {
+          for (int i = 0; i < finalizedBlockBatchSize && it.hasNext() && preBellatrix; i++) {
+            final Map.Entry<Bytes, Bytes> entry = it.next();
+            final UInt64 slot =
+                UInt64.fromLongBits(Longs.fromByteArray(entry.getKey().toArrayUnsafe()));
+            if (spec.atSlot(slot).getMilestone().isGreaterThanOrEqualTo(SpecMilestone.BELLATRIX)) {
+              preBellatrix = false;
+              LOG.info("At slot {}, detected bellatrix block, stopping direct copy", slot);
+              continue;
+            }
+            final Optional<Bytes32> maybeRoot = dao.getFinalizedBlockRootAtSlot(slot);
+            if (maybeRoot.isEmpty()) {
+              LOG.warn("Could not find block root for slot {}", slot);
+              continue;
+            }
```

Comment:
This will skip the block from a raw copy.

The next process is a copy where we have block roots etc, so we can copy via that mechanism instead, so the block isn't lost, it's just not copied as part of this process. It could potentially not even be a warning...

Reply:
changing to be debug